### PR TITLE
feat: Add support for the Discussions MFE

### PR DESCRIPTION
**Pull request by arbrandes**
Adds support for the Discussions MFE, enabling it by default.

Optional features also enabled via waffle flag:

* Learner's tab in the MFE
* Extended moderation reason codes
* Reported content email notifications to moderators
* Enable learner stats in the activity API

#### Requirements

- [x] https://github.com/openedx/frontend-app-discussions/pull/364
- [x] https://github.com/openedx/frontend-app-discussions/pull/363
- [x] https://github.com/openedx/cs_comments_service/pull/399
- [x] https://github.com/openedx/edx-platform/pull/31345

### CODE REVIEW

**Comment by arbrandes** [comment]
@regisb, I'm still waiting on the backports from master to Olive in cs_comments_service and edx-platform, but the tutor-mfe bits are good enough to review, and the MFE actually works... as far as I can tell.  I still see fixes to frontend-app-discussions coming in (I had to whip one up [myself](https://github.com/openedx/frontend-app-discussions/pull/364)), so we'll want to keep our eyes open to pick them up before release day.

File: README.rst
Comment (regisb) [inline README.rst:254]: We're starting to have so many waffles that we need to disable individually for each MFE... Could you please add `# MFE <name>` comments such that we know for which MFE each waffle need to be toggled? In the case of the discussion MFE it's fairly obvious, of course.

**Review by regisb — APPROVED** [review]
LGTM! But we agree that we are not going to merge this PR in master, right? (only in Olive or nightly)

File: README.rst
Comment (arbrandes) [reply README.rst:260]: Good idea.  See if the changes work, then I'll apply it to the other PRs.  Or, actually, I'll just wait to group all the new MFEs in one PR, once they're all approved and #69 merges.

**Comment by arbrandes** [comment]
@regisb, yes, definitely nightly!

**Comment by arbrandes** [comment]
Ok, I've tested this in nightly and olive, and it's looking good AFAICT.  We're just blocked on the edx-platform fixes.

**Comment by arbrandes** [comment]
Rebased on the new runtime config support: good enough for nightly.  Probably worth another glance, @regisb.

**Review by regisb — APPROVED** [review]
Awesome!

**Comment by arbrandes** [comment]
...aaand, we're in!

**Pull request merged**
